public/pages/Dashboard/Components/AnomaliesLiveChart.tsx

```diff
@@ -0,0 +1,293 @@
+    // clear previous result
+    const currentLiveAnomalyData = [] as object[];
+    for (let detector of currentDetectors) {
+      const resp = await dispatch(
+        getDetectorResults(detector.id, getRecentAnomalyResultQuery)
+      );
+      currentLiveAnomalyData.push({
+        detector: detector,
+        results: resp.data.response.results,
+      });
+    }
+
+    setLiveAnomalyData(currentLiveAnomalyData);
+
+    setLiveTimeRange({
+      startDateTime: moment().subtract(30, 'minutes'),
+      endDateTime: moment(),
+    });
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    async function initializeAnomaliesChart() {
+      const liveDetectors = await getLiveDetectors();
+      getLiveAnomalyData(liveDetectors);
+      setTimer(
+        setInterval(getLiveAnomalyData, MIN_IN_MILLI_SECS, liveDetectors)
+      );
+    }
+    initializeAnomaliesChart();
+    return () => {
+      clearInterval(timer);
+    };
+  }, [props.selectedDetectors]);
+
+  const timeFormatter = niceTimeFormatter([
+    liveTimeRange.startDateTime.valueOf(),
+    liveTimeRange.endDateTime.valueOf(),
+  ]);
+
+  const liveVisualizedAnomalies = liveAnomalyData.flatMap(
+    detectorAnomalyResult =>
+      visualizeDetectorAnomalyResult(detectorAnomalyResult)
+  );
+
+  const prepareVisualizedAnomalies = (
+    liveVisualizedAnomalies: object[]
+  ): object[] => {
+    // add data point placeholder at every minute,
+    // to ensure chart evenly distrubted
+    const existingPlotTimes = liveVisualizedAnomalies.map(anomaly =>
+      getFloorPlotTime(get(anomaly, AD_DOC_FIELDS.PLOT_TIME, 0))
+    );
+    const result = [...liveVisualizedAnomalies];
+
+    for (
+      let currentTime = getFloorPlotTime(liveTimeRange.startDateTime.valueOf());
+      currentTime <= liveTimeRange.endDateTime.valueOf();
+      currentTime += MIN_IN_MILLI_SECS
+    ) {
+      if (existingPlotTimes.includes(currentTime)) {
+        continue;
+      }
+      result.push({
+        [AD_DOC_FIELDS.DETECTOR_NAME]: '',
+        [AD_DOC_FIELDS.PLOT_TIME]: currentTime,
+        [AD_DOC_FIELDS.ANOMALY_GRADE]: null,
+      });
+    }
+    return result;
+  };
+
+  const timeNowAnnotation = {
+    dataValue: getFloorPlotTime(liveTimeRange.endDateTime.valueOf()),
+    header: 'Now',
+    details: moment(liveTimeRange.endDateTime).format('MM/DD/YY h:mm a'),
+  } as LineAnnotationDatum;
+
+  const annotations = [timeNowAnnotation];
+
+  return (
+    <ContentPanel
+      title="Live Anomalies"
+      titleSize="s"
+      isLive={true}
+      subTitle="Live anomaly results across detectors for the last 30 minutes"
+      subtitleClassName="live-anomaly-results-subtile"
+    >
+      <div
+        style={{
+          height: '300px',
+          width: '100%',
+          opacity: 1,
+        }}
+      >
+        {isLoading ? (
+          <EuiText>
+            <EuiTextAlign textAlign="center">
+              <p>Loading anomaly data...</p>
```

Comment:
I think we are using loading icon. https://elastic.github.io/eui/#/display/loading

Reply:
Thanks. I will add it.